### PR TITLE
fix home path resolution at data_dir creation

### PR DESCRIPTION
**Pull request by St333p**
This is my "dirty" attempt to solve the issue we discussed about in #22. I realize it's not as clean as it could be (path gets converted from String to PathBuf to String to PathBuf again), but I'm not able to come up with anything better. Still it seems to work.

### CODE REVIEW

File: src/opts.rs [hunk 1]
```diff
@@ -141,11 +141,13 @@ pub struct Opts {
 impl Opts {
     pub fn process(&mut self) {
         LogLevel::from_verbosity_flag_count(self.verbose).apply();
-
+        let me = self.clone();
+        let mut data_dir = self.data_dir.to_string_lossy().to_string();
+        me.process_dir(&mut data_dir);
+        self.data_dir = PathBuf::from(data_dir);
```
Comment (dr-orlovsky) [inline src/opts.rs:147]: I think we can go for

```suggestion
        let mut me = self.clone();
        me.data_dir = shellexpand::tilde(me.data_dir).to_string();
```

No need to use `process_dir` which also does `{data_dir}` substitution and opens attack vector with the call stack overflow

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
One suggestion, otherwise LGTM

File: src/opts.rs (hunk 1 above)
Comment (St333p) [reply src/opts.rs:147]: I don't know why I stuck with using `process_dir`, like that it's definitely better.

Comment (St333p) [reply src/opts.rs]: I needed to do some type conversions (in e4f7d13) to get it compile.

**Pull request merged**